src/video/vga_palette.c

```diff
@@ -21,9 +22,10 @@ void vga_palette_tint_range(vga_palette *pal, vga_index ref_index, vga_index sta
 }
 
 void vga_palette_mix_range(vga_palette *pal, vga_index ref_index, vga_index start, vga_index end, uint8_t step) {
+    assert(end + (vga_index)255 >= start);
     vga_color ref = pal->colors[ref_index];
     uint32_t inv = 255 - step;
-    for(vga_index i = start; i < end; i++) {
+    for(vga_index i = start; i != end; i++) {
```

Comment:
I wonder, why the '!=' and not '<' ?

Reply:
Kreissack calls this function with (1, 1 + 255).
1 + 255 is 0, in vga_indexes.

Although we could use a wider integer type for these indexes, i prefer this solution because we don't need to worry about overflowing the pal->colors array.

Reply:
I see, it's probably my mistake from when I moved these here. Okay, I'm fine with either, though a code comment might clear things for future generations.

Reply:
Added a comment about the inequality